enable_retired_username_email/signals.py

```diff
@@ -0,0 +1,72 @@
+"""
+Signal handler for setting setting modified username and email hash in user model.
+"""
+
+import time
+
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from enable_retired_username_email.compat import get_retired_email_hash, get_user_retirement_status_model
+
+UserRetirementStatus = get_user_retirement_status_model()
+
+
+@receiver(post_save, sender=UserRetirementStatus)
+def replace_username_email_hash(
+    instance: UserRetirementStatus,
+    created,
+    **kwargs
+):  # pylint: disable=unused-argument
+    """
+    This function is triggered when UserRetirementStatus is saved.
+
+    Upon receiving the post_save
+    signal for the UserRetirementStatus model,
+    appends to the original username and email and
+    calculates their hash values. These are then
+    set to the User model and the corresponding
+    UserRetirementStatus is deleted
+    """
+    if created:
```

Comment:
I have an issue here theoretically, if you have the `created` check this means the block of code will be executed multiple times, i.e every time status changes but we just need to generate this one time right?

If we just check for `PENDING` status and at that point, if we generate the retirement username then we can easily avoid multiple unnecessary save calls.

What do you think?

Reply:
The `created` flag is `True` only once when a new entry is created for the UserRetirementStatus model. For any subsequent changes to the entry including change in status, the `created` flag is `False`.

Reply:
alright, make sense